AUTHORS.txt

```diff
@@ -225,6 +225,11 @@ Pioneer includes the following third-party software:
   Copyright 2014 Melissa O'Neill <oneill@pcg-random.org>
   Licensed under the Apache License, Version 2.0
 
+  fast_float
```

Comment:
No way to do this without bringing in a another thirdparty library?

Reply:
No. To my knowledge, no standard library method (even including istringstream) allows reading from a size-delimited rather than null-terminated string - and especially without copying the file data to an owned buffer! Additionally, fast_float is faster than the standard library methods and cross-platform deterministic; i.e. the same input string will give the same bit-exact value on any hardware with IEEE754 semantics.

I could not have used the JSON parser's string-to-double parsing implementation, as that's buried in its lexer and just calls std::strtod. I admittedly only confirmed this just now.

I don't necessarily consider this a bad dependency to add, though I do lament that we have to pull in a dependency just to get a non-null string parsed.